test/functional/homepage_controller_test.rb

```diff
@@ -5,6 +5,68 @@ class HomepageControllerTest < ActionController::TestCase
     login_as_stub_user
   end
 
+  context "#confirm_destroy" do
+    should "render confirm destroy page for draft edition" do
+      popular_links = FactoryBot.create(:popular_links, state: "draft")
+
+      get :confirm_destroy, params: { id: popular_links.id }
+
+      assert_template "homepage/popular_links/confirm_destroy"
+    end
+
+    should "not render confirm destroy page if edition is published" do
+      popular_links = FactoryBot.create(:popular_links, state: "published")
+
+      get :confirm_destroy, params: { id: popular_links.id }
+
+      assert_redirected_to show_popular_links_path
+    end
+  end
+
+  context "destroy" do
+    setup do
+      @popular_links = FactoryBot.create(:popular_links, state: "draft")
+      get :confirm_destroy, params: { id: @popular_links.id }
+    end
+
+    should "delete the popular links from the database" do
+      assert_equal 1, PopularLinksEdition.count
+
+      delete :destroy, params: { id: @popular_links.id }
+
+      assert_equal 0, PopularLinksEdition.count
+    end
+
+    should "display a success message if successful" do
```

Comment:
may be can merge this test with the previous one to say "delete the popular links from the database and display a success message".
Because, there is nothing in the second test showing that delete was succesful.